tfexec/version.go

```diff
@@ -0,0 +1,78 @@
+package tfexec
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+)
+
+// Version returns structured output from the `terraform version` command including both the Terraform CLI version
+// and any initialized provider versions.
+func (tf *Terraform) Version(ctx context.Context) (tfVersion *version.Version, providerVersions map[string]*version.Version, err error) {
+	// TODO: support re-invocation of this?
+	return tf.execVersion, tf.provVersions, nil
+}
+
+func (tf *Terraform) version(ctx context.Context) (*version.Version, map[string]*version.Version, error) {
+	versionCmd := tf.buildTerraformCmd(ctx, "version")
+	var errBuf strings.Builder
+	var outBuf bytes.Buffer
+	versionCmd.Stderr = &errBuf
+	versionCmd.Stdout = &outBuf
+
+	err := versionCmd.Run()
+	if err != nil {
+		fmt.Println(errBuf.String())
+		return nil, nil, fmt.Errorf("unable to run version command: %w, %s", err, errBuf.String())
+	}
+
+	tfVersion, providerVersions, err := parseVersionOutput(outBuf.String())
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to parse version: %w", err)
+	}
+
+	return tfVersion, providerVersions, nil
+}
+
+var (
+	simpleVersionRe = `v?(?P<version>[0-9]+(?:\.[0-9]+)*(?:-[A-Za-z0-9\.]+)?)`
+
+	versionOutputRe         = regexp.MustCompile(`^Terraform ` + simpleVersionRe)
+	providerVersionOutputRe = regexp.MustCompile(`(\n\+ provider[\. ](?P<name>\S+) ` + simpleVersionRe + `)`)
+)
+
+// TODO: maybe add JSON output to the version command to simplify all of this?
```

Comment:
This is done in https://github.com/hashicorp/terraform/pull/25252 and will land in 0.13. https://github.com/hashicorp/terraform-exec/issues/13 tracks use of this feature in tfexec.

```suggestion
```